packages/labs/preact-signals/src/test/signal-watcher_test.ts

```diff
@@ -97,4 +97,72 @@ suite('SignalWatcher', () => {
     assert.equal(el.shadowRoot?.querySelector('p')?.textContent, 'count: 3');
     assert.equal(readCount, 2);
   });
+
+  test('type-only test where mixin on an abstract class preserves abstract type', () => {
+    if (true as boolean) {
+      // This is a type-only test. Do not run it.
+      return;
+    }
+    abstract class BaseEl extends LitElement {
+      abstract foo(): void;
+    }
+    // @ts-expect-error foo() needs to be implemented.
+    class TestEl extends SignalWatcher(BaseEl) {}
+    console.log(TestEl); // usage to satisfy eslint.
+
+    const TestElFromAbstractSignalWatcher = SignalWatcher(BaseEl);
+    // @ts-expect-error cannot instantiate an abstract class.
+    new TestElFromAbstractSignalWatcher();
+
+    // This is fine, passed in class is not abstract.
+    const TestElFromConcreteClass = SignalWatcher(LitElement);
+    new TestElFromConcreteClass();
+  });
+
+  test('can mixin an abstract class and watch a signal', async () => {
```

Comment:
abstract classes compile to regular classes in JS so I think type checking is good enough if other tests already cover runtime behavior with classes.

Reply:
Great point. I'll remove this test!

Reply:
Done